### PR TITLE
Updates the changelog and version number for the 0.5.0 release

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:**

* Version number updated to 0.5.0

* Changelog updated

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -19,7 +14,10 @@
 
 ### Breaking changes
 
-* The method `Device.supported` was removed.
+* The method `Device.supported` was removed. The replacement methods are
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
* The method `Device.supported` was removed, and replaced with the methods
```

**Comment by josh146** [comment]
Have added in the missing PRs! Thanks @co9olguy

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -45,26 +72,38 @@
...
 * Changed the signature of the `RandomLayer` and `RandomLayers` templates to have a fixed seed by default.
   [#258](https://github.com/XanaduAI/pennylane/pull/258)
 
+* `setup.py` has been cleaned up, removing the non-working shebang,
+  and removing unused imports.
+  [#262](https://github.com/XanaduAI/pennylane/pull/262)
+
+### Documentation
+
+* Fixes a typo in `QubitStateVector`.
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
* Fixed a typo in `QubitStateVector`.
```

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -45,26 +72,38 @@
...
+  [#295](https://github.com/XanaduAI/pennylane/pull/296)
+
+* Fixes a typo in the `default_gaussian.gaussian_state` function.
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
* Fixed a typo in the `default_gaussian.gaussian_state` function.
```

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -45,26 +72,38 @@
...
+  [#293](https://github.com/XanaduAI/pennylane/pull/293)
+
+* Fixes a typo in the gradient recipe within the `RX`, `RY`, `RZ`
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
* Fixed a typo in the gradient recipe within the `RX`, `RY`, `RZ`
```

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -45,26 +72,38 @@
...
+  operation docstrings.
+  [#248](https://github.com/XanaduAI/pennylane/pull/248)
+
+* Fixes a broken link in the tutorial documentation, as a
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
* Fixed a broken link in the tutorial documentation, as a
```

**Pull request merged**